linera-base/src/data_types.rs

```diff
@@ -842,21 +863,77 @@ impl HashedBlob {
     }
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(rename = "HashedBlob")]
+struct SerializableHashedBlob {
+    pub id_type: String,
+    pub blob: Blob,
+}
+
 impl Serialize for HashedBlob {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
-        S: serde::Serializer,
+        S: Serializer,
     {
-        self.blob.serialize(serializer)
+        let id_type = match self.id {
+            BlobId::Data(_) => "User",
+        }
```

Comment:
Why are we managing the serialization of the enum ourself?

Reply:
Will change the type to be an enum